API-strategie-modulen/API-strategie-mod-geo/request-response.md

```diff
@@ -101,6 +97,90 @@ However, until the filtering module is written, the geospatial module retains ru
+      "coordinates": [5.2795,52.1933]
+    },
+    "properties": {
+      "naam": "Paleis Soestdijk",
+      ...
+    }
+  }
+  </pre>
+  Example: POST feature collection
+  <pre>
+  {
+    "type": "FeatureCollection",
+    "features": [
+    {
+      "type": "Feature",
+      "geometry":  {
+        "type": "Point",
+        "coordinates": [5.2795,52.1933]
+      },
+      "properties": {
+        "naam": "Paleis Soestdijk",
+        ...
+      }
+    }]
+  }  
+  </pre>
+  <h4 class="rulelab">How to test</h4>
+  <ul>
+    <li>Create a new resource that includes feature content (i.e. coordinates) using the HTTP POST method with request media type <code>application/geo+json</code> in the <code>Content-Type</code> header.</li>
+    <li>Validate that a response with status code <code>201</code> (Created) is returned.</li>
+    <li>Validate that the response includes the <code>Location</code> header with the URI of the newly added resource.
+  </ul>
+</div>
+
+In case a REST API does not have to comply to the OGC API Features specification, e.g. for usage in administrative applications, the REST API shall use the JSON data format. If a resource contains geometry, that geometry shall be embedded as a GeoJSON <code>Geometry</code> object within the resource. The media type <code>application/json</code> must be supported. This may also apply to other media types <code>application/*+json</code>, however this depends on the media type specification. If the media type specification prescribes that resource information must be embedded in a JSON structure defined in the media specification, then the media type should not be supported while it is impossible to comply to that specification with the method described below. The media type <code>application/geo+json</code> should not be supported while the resource does not comply to the GeoJSON specification, i.e. the request resource does not embed a feature or feature collection.
+A template for the definition of the schemas for the GeoJSON <code>Geometry</code> object in the requests in OpenAPI definitions are available [geometryGeoJSON.yaml](https://schemas.opengis.net/ogcapi/features/part1/1.0/openapi/schemas/geometryGeoJSON.yaml).
+In case a collection of resources is embedded in the request resource, the name of the array containing the resources should be the plural of the resource name.
+
+<div class="rule" id="api-geo-4">
+  <p class="rulelab"><strong>API-GEO-4</strong>: Embed GeoJSON <code>Geometry</code> object as part of the JSON resource</p>
+  <p>When a JSON (<code>application/json</code>) request contains a geometry, represent it in the same way as the <code>Geometry</code> object of GeoJSON.</p>
+  Example: POST resource containing geometry
+  <pre>
+  {
+    "type": "Feature",
+    "naam": "Paleis Soestdijk",
+    "geometrie":  {
+      "type": "Point",
+      "coordinates": [5.2795,52.1933]
+    }
+  }
+  </pre>
+  Example: POST resource containing geometry collection
```

Comment:
Als hier een resource met een geometry collection wordt gecreëerd, dan klopt het voorbeeld niet, want daarin wordt een Point meegegeven. Het voorbeeld bevat wel een array van resources die een geometrie bevatten, dus ik denk dat de beschrijving hier niet juist is.

Het zou wel goed zijn ook een voorbeeld toe te voegen waarbij één resource met meerdere geometrieën wordt aangemaakt, in dat geval wordt er een geometrieCollection meegegeven.

Reply:
Ik zal het aanpassen naar een resource met als geometrie een GeometryCollection. Een POST met daarin een array van resources is niet volgens de definitie van REST, dus dit voorbeeld klopt sowieso niet.